### ISSUE TITLE
Add syntactic sugar for tuples

### ISSUE THREAD

**Issue by brendanzab** [gluon-lang/gluon#88]
So in SML, tuples desugar into records with numeric fields consecutively increasing from `1`. Ie.

```
$ sml
Standard ML of New Jersey v110.78 [built: Thu Aug 20 19:23:18 2015]
- {1 = 2, 2 = 3} : int * int;
val it = (2,3) : int * int
- (2, 3) : {1 : int, 2 : int};
val it = (2,3) : int * int
```

Not sure if we would want to go down this route, or add them as a separate type.

In Idris, tuples [are sugar for nested `Pair`s](http://docs.idris-lang.org/en/latest/tutorial/typesfuns.html#tuples), so you can implement things over them generically:

```
Idris> (1, (MkPair 2 3))
(1, 2, 3) : (Integer, Integer, Integer)
Idris> (1, (2, 3))
(1, 2, 3) : (Integer, Integer, Integer)
Idris> (1, (MkPair 2 3))
(1, 2, 3) : (Integer, Integer, Integer)
Idris> :doc Pair
Data type Builtins.Pair : (A : Type) -> (B : Type) -> Type
    The non-dependent pair type, also known as conjunction.
    Arguments:
        A : Type  -- the type of the left elements in the pair

        B : Type  -- the type of the left elements in the pair

Constructors:
    MkPair : (a : A) -> (b : B) -> (A, B)
        A pair of elements
        Arguments:
            a : A  -- the left element of the pair

            b : B  -- the right element of the pair
```

**Comment by Marwes** [comment]
I will just add that Rust tuples are currently mapped to records with the fields `{ _0: ..., _1: ..., _2: ... }`. This allows indexing on 'tuples' but may not be intuitive. Rust allows tuples to be indexed directly as `t.0` but then you run in to problems when parsing things such as `t.0.1` which have to be written as `(t.0).1`

**Comment by brendanzab** [comment]
Yeah. SML uses `#label`, which removes the integer ambiguity, eg. `#2 (1, 2)` - but I think we want to stick to using the dot operator for record projection.

Another problem with using natural indexes would be:

```haskell
let { 0 = x, 1 = y } = (1, 2);
```

**Comment by Marwes** [comment]
Implemented in #265
